master/getting-started/kubernetes/upgrade.md

```diff
@@ -167,3 +172,31 @@ To upgrade the policy controller:
 We recommend running the policy controller as a Kubernetes Deployment with type "recreate", in which
 case upgrade can be handled entirely through the
 standard [Deployment mechanism](http://kubernetes.io/docs/user-guide/deployments/#updating-a-deployment)
+
+## Upgrading to v1 NetworkPolicy semantics
+
+Calico v2.3.0 when using the Kubernetes datastore driver interprets the Kubernetes `NetworkPolicy` differently
+than previous releases, as specified in [upstream Kubernetes](https://github.com/kubernetes/kubernetes/pull/39164#issue-197243974).
+
+To maintain behavior when upgrading to Calico v2.3.0+, you should follow the following steps.
+
+- In any Namespace that previously did _not_ have a "DefaultDeny" annotation:
+  - Delete any NetworkPolicy objects in that Namespace.  After upgrade, these policies will become active and my block traffic that was previously allowed.
+- In any Namespace that previously had a "DefaultDeny" annotation:
+  - Create a NetworkPolicy which matches all pods but does not allow any traffic.  After upgrade, the Namespace annotation will have no effect, but this empty NetworkPolicy will provide the same behavior.
+
+Here is an example of a NetworkPolicy which selects all pods in the Namespace, but does not allow any traffic:
+
+```yaml
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: default-deny
+spec:
+  podSelector:
+```
+
+> **Note**:
+>
+> The above steps should be followed when upgrading to Calico v2.3.0+ using the Kubernetes
+> datastore driver, independent of the Kubernetes version being used.
```

Comment:
If you are using the Kubernetes datastore driver, you should perform the steps above when upgrading to Calico v2.3.0+, regardless of the Kubernetes version.
You do NOT need to perform the above steps if you are using the etcd datastore driver (in v2.3.x - the corresponding change is planned for the etcd datastore driver in a v2.4.x release)